twoliter/src/variant.rs

```diff
@@ -0,0 +1,46 @@
+use crate::common::exec;
+use anyhow::{Context, Result};
+use serde_json::Value;
+use std::path::Path;
+use tokio::process::Command;
+
+pub(crate) async fn variant_version<S>(variants_dir: &Path, variant: S) -> Result<String>
+where
+    S: AsRef<str>,
+{
+    serde_json::from_str::<Value>(
+        &exec(
+            Command::new("cargo").args([
+                "metadata",
```

Comment:
@bcressey we discussed a few ideas about where we should put image (i.e. variant) versions. There could be a single one for all variants in Twoliter.toml, variants could be enumerated and versioned in Twoliter.toml, we could put it in the variant's custom Cargo.toml metadata...

Here we are using the actual cargo package version as the variant's version. What do you think?

Note, we are doing it now because it removes the Release.toml requirement.

Reply:
> Here we are using the actual cargo package version as the variant's version. What do you think?

My biggest gripe is that it's inconsistent with the cargo manifests for packages, where the crate version is ignored and only the version in the package's spec file has any external effect. As a minor nit, I also don't love parsing the crate manifest in different ways with different tools; we already have a parser in buildsys and could factor that out into a standalone crate if we needed to.

A single version in `Twoliter.toml` is the most consistent with existing logic and would make what I consider to be the common case easier - a project where multiple variants are built and released together, all tied to a specific Git commit tagged with a version. Updating a dozen versions across all variant manifests would be a lot of boilerplate for a routine chore, vs. just editing the top-level file.

I am still somewhat interested in the direction where variants (and kits) all have associated RPM spec files, to include package dependencies via Requires and stray files that don't make sense elsewhere. If we did that we could get the variant's version from the spec file, which would be consistent with package versioning.

But I'm not convinced that would be better. Before we added `application-inventory.json`, none of these internal versions had any external significance. The spec file versions were mostly there as an implementation detail and a nod to spec file conventions. They're also a bit annoying to maintain since every package update requires at least two files to edit.

My preference would be to start with a top-level `release-version` field, and if we need variant-specific versions for some reason, we could add a `release-version-override` field to `package.metadata.build-variant`.